components/FacetMenu/DatasetFacetMenu.vue

```diff
@@ -35,17 +63,24 @@ export default {
       type: Object,
       default: () => {}
     },
-    defaultCheckedFacetIds: {
-      type: Array,
-      default: () => []
-    }
   },
 
   data() {
     return {
       showFacetMenu: (process.env.show_facet_menu == 'true') ? true : false,
       selectedFacets: {},
-      selectedFacetArray: []
+      selectedFacetArray: [],
+      defaultCheckedFacetIds: [],
+      embargoedFacetCategory : embargoedFacetCategory,
+      embargoFacetCategoryTooltip: embargoFacetCategoryTooltip,
+      numKeys: 0,
+      latestUpdateKey: '',
+    }
+  },
+
+  mounted() {
+    if (this.$route.query.selectedFacetIds) {
+      this.defaultCheckedFacetIds = this.$route.query.selectedFacetIds.split(",")
```

Comment:
Is it correct to say that the facet ids may not be consistent if facets have been removed, added or the order has changes? 
The link may not produce the same facets filter?

Reply:
hmm the facets should stay consistent, but they may not be visible eventhough they may be selected which is fine. We always populate all the facets that we retrieve from Algolia, but we decide which ones to show via the visibleFacets prop.

I think you are correct that they could get out of sync if we did manipulate the facet prop directly, but that should not happen. The facets get populated via a call to Algolia and these facets should always have the same id's since results from algolia are returned in a specific order (except for when the backend changes. For example, lets say a user selects 'Andipose Tissue' then the route would be '/data?type=dataset&selectedFacetIds=0'. However if we update the backend to have another facet called 'ACL' then Andipose tissue would now have an id of 1 (Andipose alphabetically comes after ACL) when we populate the facets so if the user attempted to navigate directly to '/data?type=dataset&selectedFacetIds=0' after the backend changes then 'ACL' would be selected instead of 'Andipose Tissue'. Could this be an issue @jwagenaar or is this an acceptable behavior? I could see it being confusing if someone copy/pasted a link and then navigated to it in the future)

Reply:
Hmm, yeah that is not ideal. I think mostly because we also want to use this to link the items on the front-page to a specific search. If we didn't do that, I don't think it matters too much but if we hard-code the organ button on the sparc front-page, this is most likely going to fail.

Reply:
that said, the alternative of providing the full path to the attribute is also far from ideal...

Reply:
maybe would use Vuex instead of url to capture state

Reply:
I discussed this with Joost and it was decided to implement this using Vuex as tracked by this ticket: https://www.wrike.com/workspace.htm?acc=3203588#/inbox/task/760474893

Note that as it is now it will work with the organ icons on the front page because the ids are calculated dynamically, however, any copied link may become out of date when the backend changes so it was decided to track facet menu state in Vuex instead of the url.

Reply:
Vuex store lasts for a session, all data would be wiped on refresh and would not allow the users to go back to previous selected facets like an URL would. Is it not possible to have Algolia to couple each facet with an unique ID that would never change?

Reply:
I did not realize a page refresh wipes it. I believe each facet having a unique id is possible, just not sure how it would be organized in Algolia. I would have to talk to Jeff about this if this is the route we decide to take. Otherwise, it might make sense if a page refresh resets the selections, so long as we still persist selections between tabs. I'll bring it up with @jwagenaar next time I talk to him